src/hooks/use-posthog-analytics.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
+
+declare global {
+	interface Window {
+		posthog?: $TSFixMe
+	}
+}
+
+function onRouteChangeComplete() {
+	// PostHog documentation for capturing pageviews in SPA with the JS web installation:
+	// https://posthog.com/docs/libraries/js#single-page-apps-and-pageviews
```

Comment:
⛏️ Nitpick: I was at first confused how we were capturing `pageview` events on load. The docs link definitely helped! Felt like it may be useful to surface the "PostHog captures stuff automatically" directly as well:

```suggestion
	/**
	 * PostHog automatically captures a `pageview` for initial page loads.
	 * Subsequent client-side navigation events have to be captured manually,
	 * which is why we have to set up this `onRouteChangeComplete` event.
	 * 
	 * PostHog documentation for capturing pageviews in SPA with the JS web installation:
	 * https://posthog.com/docs/libraries/js#single-page-apps-and-pageviews
	 */
```